[Glossary] Adding the definiton of vCPU to the Glossary (#1975)
cntt-n/CNTT: doc/common/glossary.md

<a name="1.1"></a>
## Software Layer Terminology

- **Cloud Infrastructure:** A generic term covering **NFVI**, **IaaS** and **CaaS** capabilities - essentially the infrastructure on which a **Workload** can be executed.
>_*Note:*_ **NFVI**, **IaaS** and **CaaS** layers can be built on top of each other. In case of CaaS some cloud infrastructure features (e.g.: HW management or multitenancy) are implemented by using an underlaying **IaaS** layer.

> - **Cloud Infrastructure Profile:** The combination of the Cloud Infrastructure Software Profile and the Cloud Infrastructure Hardware Profile that defines the capabilities of the Cloud Infrastructure.

Reviewer: I think we may need to be specify the nature of capabilities and for what; the profile also specifies certain configurations. While Cloud Infrastructure Software Profile is defined the Hardware Profile is not.
```suggestion
- **Cloud Infrastructure Profile:** The combination of the Cloud Infrastructure Software Profile and the Cloud Infrastructure Hardware Profile that defines the capabilities and configuration of the Cloud Infrastructure resources made available to workloads.
```

- **Cloud Infrastructure Software Configuration:** a set of settings (Key:Value) that are applied/mapped to **cloud infrastructure** SW deployment.

> - **Cloud Infrastructure Software Profile:** defines the behaviour, capabilities and metrics provided by a Cloud Infrastructure Software Layer.

Reviewer: ```suggestion
- **Cloud Infrastructure Software Profile:** defines the behaviour, capabilities and metrics provided by a Cloud Infrastructure Software Layer on resources made available to workloads.
```

 >_*Note:*_ The NFVI can span across many locations, e.g. places where data centres or edge nodes are operated. The network providing connectivity between these locations is regarded to be part of the cloud infrastructure. **NFVI** and **VNF** are the top-level conceptual entities in the scope of Network Function Virtualisation. All other components are sub-entities of these two main entities.

- **Network Service (NS):** composition of **Network Function**(s) and/or **Network Service**(s), defined by its functional and behavioural specification, including the service lifecycle.

- **Software Defined Storage (SDS):** An architecture which consists of the storage software that is independent from the underlying storage hardware. The storage access software provides data request interfaces (APIs) and the SDS controller software provides storage access services and networking.

- **Virtual Application (VA):** A general term for software which can be loaded into a Virtual Machine.
  >_*Note:*_ a **VNF** is one type of VA.

> - **Virtual CPU (vCPU):** Represents a portion of the host's computing resources allocated to a workload. One or more vCPUs can be assigned to a workload.

Reviewer: technically the vCPUs are assigned to a VM or container (or a virtualised resource).
```suggestion
- **Virtual CPU (vCPU):** Represents a portion of the host's computing resources allocated to a virtualised resource (virtual machine or container). One or more vCPUs can be assigned to a  virtualised resource.
```

Reviewer 2: @EditKoselak and @pgoyal01: i agree with the last version.  @EditKoselak: can you please accept the update or respond?

- **Workload:** Workload refers to software running on top of compute resources such as **VMs** or **Containers**. Most relevant workload categories in context of cloud infrastructure are:
  - **Data Plane Workloads:** related to packet handling of the end-to-end communication between applications. These tasks are expected to be very I/O and memory read/write operations intensive.
  - **Control Plane Workloads:** that perform tasks related to any other communication between NFs that is not directly related to the end-to-end data communication between applications. For example, this category includes session management, routing or authentication.
  - **Storage Workloads:** that perform tasks related to disk storage (either SSD or HDD or other).  Examples range from non-intensive router logging to more intensive database read/write operations.

<a name="1.2"></a>
## Hardware Layer Terminology

- **Cloud Infrastructure Hardware Configuration:** a set of settings (Key:Value) that are applied/mapped to **Cloud Infrastructure** HW deployment.

> - **Cloud Infrastructure Hardware Profile:** defines the behaviour, capabilities, configuration, and metrics provided by a cloud infrastructure hardware layer.

Reviewer: ```suggestion
- **Cloud Infrastructure Hardware Profile:** defines the behaviour, capabilities, configuration, and metrics provided by a cloud infrastructure hardware layer resources made available to workloads.
```

Reviewer 2: @EditKoselak and @pgoyal01: i agree with the last version.  @EditKoselak: can you please accept the update or respond?

- **Cloud Infrastructure:** A generic term covering **NFVI**, **IaaS** and **CaaS** capabilities - essentially the infrastructure on which a **Workload** can be executed.
>_*Note:*_ **NFVI**, **IaaS** and **CaaS** layers can be built on top of each other. In case of CaaS some cloud infrastructure features (e.g.: HW management or multitenancy) are implemented by using an underlaying **IaaS** layer.

- **Cloud Infrastructure Profile:** The combination of the Cloud Infrastructure Software Profile and the Cloud Infrastructure Hardware Profile that defines the capabilities and configuration of the Cloud Infrastructure resources available for the workloads.

- **Cloud Infrastructure Software Configuration:** a set of settings (Key:Value) that are applied/mapped to **cloud infrastructure** SW deployment.

> - **Cloud Infrastructure Software Profile:** defines the behaviour, capabilities and metrics provided by a Cloud Infrastructure Software Layer on resources available for the workloads.

Reviewer 3: Why do we need to add this? The infrastructure defines that for any resources, also if they are unavailable.

Reviewer: @ulikleber Is the concern about "available" then that can be dropped -- is "... on resources for workloads" better?

> - **Cloud Native Network Function (CNF):** A cloud native network function (CNF) is a cloud native application that implements network functionality. A CNF consists of one or more microservices. All layers of a CNF is developed using Cloud Native Principles including immutable infrastructure, declarative APIs, and a “repeatable deployment process”.

Reviewer 3: All layers ... is ??? 
It should stay "are".

- **Virtual Application (VA):** A general term for software which can be loaded into a Virtual Machine.
  >_*Note:*_ a **VNF** is one type of VA.

> - **Virtual CPU (vCPU):** Represents a portion of the host's computing resources allocated to a virtualised resource, for example, to a virtual machine or a container. One or more vCPUs can be assigned to a  virtualised resource.

Reviewer 3: This definition is too wide. It applies to many other things, including a physical CPU in an SMP system or a Linux process.

Reviewer 4: Any suggestions how to make it more narrow?

- **Cloud Infrastructure Hardware Configuration:** a set of settings (Key:Value) that are applied/mapped to **Cloud Infrastructure** HW deployment.

> - **Cloud Infrastructure Hardware Profile:** defines the behaviour, capabilities, configuration, and metrics provided by a cloud infrastructure hardware layer resources available for the workloads.

Reviewer 3: Why do we need to add this? The hardware profile defines that for any resources, also if they are unavailable.

Reviewer: @ulikleber  is "... on resources for workloads" better?
